vividus-bdd-engine/src/main/java/org/vividus/bdd/model/jbehave/Scenario.java

```diff
@@ -92,4 +100,72 @@ public void setEnd(long end)
                        .filter(Objects::nonNull)
                        .orElse(List.of());
     }
+
+    /**
+     * Get unique <b>meta</b> value
+     *
+     * <p>If the <b>meta</b> does not exist or has no value, an empty {@link Optional} will be returned
+     *
+     * <p><i>Notes</i>
+     * <ul>
+     * <li><b>meta</b> value is trimmed upon returning</li>
+     * <li><i>;</i> char is used as a separator for <b>meta</b> with multiple values</li>
+     * </ul>
+     *
+     * @param metaName the meta name
+     * @return the meta value
+     * @throws NotUniqueMetaValueException if the <b>meta</b> has more than one value
+     */
+    public Optional<String> getUniqueMetaValue(String metaName) throws NotUniqueMetaValueException
+    {
+        Set<String> values = getMetaValues(metaName);
+        if (values.size() > 1)
+        {
+            throw new NotUniqueMetaValueException(metaName, values);
+        }
+        return values.isEmpty() ? Optional.empty() : Optional.of(values.iterator().next());
+    }
+
+    /**
+     * Get all <b>meta</b> values
+     *
+     * <p><i>Notes</i>
+     * <ul>
+     * <li><b>meta</b>s without value are ignored</li>
+     * <li><b>meta</b> values are trimmed upon returning</li>
+     * <li><i>;</i> char is used as a separator for <b>meta</b> with multiple values</li>
+     * </ul>
+     *
+     * @param metaName the meta name
+     * @return  the meta values
+     */
+    public Set<String> getMetaValues(String metaName)
+    {
+        return getMetaStream().filter(m -> metaName.equals(m.getName()))
+                              .map(Meta::getValue)
+                              .filter(StringUtils::isNotEmpty)
+                              .map(String::trim)
+                              .map(value -> StringUtils.splitPreserveAllTokens(value, META_VALUES_SEPARATOR))
+                              .flatMap(Stream::of)
+                              .map(String::trim)
+                              .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    /**
+     * Determine if scenario has <b>meta</b> with the name
+     *
+     * @param metaName the meta name
+     * @return {@code true} if scenario has meta with the name
+     */
+    public boolean hasMetaWithName(String metaName)
+    {
+        return getMetaStream().anyMatch(m -> metaName.equals(m.getName()));
+    }
+
+    private Stream<Meta> getMetaStream()
+    {
+        return Optional.ofNullable(getMeta())
```

Comment:
Does it makes sense to bake meta into something like this Map<String, List<String>> to avoid re-scanning

Reply:
- I would prefer to keep this POJO mappable to JSON generated by jbehave
- this avoid re-scanning is coupling to importers logic, but this POJO in common module and you don't know how the meta can be used in the future
- not sure that it's adding new deserializer, the performance gain is too low and not proven

Reply:
1. There is no need to touch the field
2. The meta is a key and a value currently you need to do something like this `getMetaStream().filter(m -> metaName.equals(m.getName()))` or this `getMetaStream().anyMatch(m -> metaName.equals(m.getName()))`

Reply:
1 then why do we need to keep duplicate List<Meta> and Map<String, List>
2 and ? :)
The meta object in JSON is not something that can be converted into Map<String, List> by jackson, that's why we have to add our custom deserialiser